### PR TITLE
Complete installation of marketplace dotorg paid plugins

### LINKED ISSUE #68610: Complete installation of dotorg managed plugins
When we Install Visual Composer, Calypso can't confirm it's installed due to the difference between the real `visualcomposer` and the current `visualcomposer-wpcom` slugs:

image

### PR DESCRIPTION
**Pull request by paulopmt1**
### Proposed Changes

Fixed thank you page for marketplace dotorg plugins like the paid Visual Composer

### Testing Instructions

- Make sure you don't have Visual Composer installed yet
- Install the [Visual Composer wpcom](http://calypso.localhost:3000/plugins/visualcomposer-wpcom) plugin at your site
- Make sure the thank you page confirms the plugin installation:

	image

- Make sure wccom marketplace plugins like `WooCommerce Subscriptions` also confirm installation as expected
- Make sure dotorg plugins like `Booking Calendar` also confirm installation as expected

Related to https://github.com/Automattic/wp-calypso/issues/68610

### CODE REVIEW

File: client/lib/plugins/utils.js [hunk 1]
```diff
@@ -104,6 +104,7 @@ export function getAllowedPluginData( plugin ) {
 		'name',
 		'network',
 		'num_ratings',
+		'org_slug',
```
Comment (DustyReagan) [inline client/lib/plugins/utils.js:107]: [I'm not 100% sure we need `org_slug`](https://github.com/Automattic/wp-calypso/pull/68602/files#r988230150), but I don't think there's any harm in including it.

**Review by DustyReagan — APPROVED** [review]
I tested with a handful of free and paid plugins. It's working great! 👍

**Review by lupus2k — APPROVED** [review]
Works like a charm. Good job @paulopmt1

File: client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx [hunk 2]
```diff
@@ -56,7 +56,14 @@ const MarketplaceThankYou = ( { productSlug }: { productSlug: string } ) => {
 	const siteId = useSelector( getSelectedSiteId );
 	const siteSlug = useSelector( getSelectedSiteSlug );
 	const isRequestingPlugins = useSelector( ( state ) => isRequesting( state, siteId ) );
-	const pluginOnSite = useSelector( ( state ) => getPluginOnSite( state, siteId, productSlug ) );
+
+	// retrieve WPCom plugin data
+	const { data: wpComPluginData } = useWPCOMPlugin( productSlug );
+	const softwareSlug = wpComPluginData
+		? wpComPluginData.software_slug || wpComPluginData.org_slug || wpComPluginData.slug
```
Comment (cpapazoglou) [inline client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx]: `|| wpComPluginData.slug` is this a fallback? Cause we have already migrated all `slug` to `software_slug` in `Marketplace_Product_Catalog` and I think we can safely remove it.

Comment (paulopmt1) [reply client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx]: Yeah, it was meant to be a fallback, but good to know. I'll remove it, thanks!

File: client/lib/plugins/utils.js (hunk 1 above)
Comment (paulopmt1) [reply client/lib/plugins/utils.js:107]: image

Currently, the `org_slug` is important to plugins like Visual Composer, which don't have `software_slug`. Do you believe we're going to change that?

Comment (cpapazoglou) [reply client/lib/plugins/utils.js:107]: Currently we use `org_slug` for visual composer. A next iteration would be looping through all requirements to check whether they are installed. In any case, we need `org_slug` whitelisted for other cases too.

Comment (paulopmt1) [reply client/lib/plugins/utils.js:107]: Thanks for the additional context. Merging it.

**Pull request merged**